core/src/main/java/com/google/cloud/sql/core/CoreSocketFactory.java

```diff
@@ -452,3 +497,37 @@ public static void setApplicationName(String applicationName) {
     System.setProperty(USER_TOKEN_PROPERTY_NAME, applicationName);
   }
 }
+
+/**
+ * This Factory generates the credentials based on the credentials read from file
+ */
+class CredsJsonFileCredentialFactory implements CredentialFactory {
+  private static final Logger logger = Logger.getLogger(CredsJsonFileCredentialFactory.class.getName());
+  private String credsJsonFileLocation;
+
+  @Override
+  public void getFileLoc(String credsJsonFileLoc)
+  {
+    this.credsJsonFileLocation = credsJsonFileLoc;
```

Comment:
This seems more like a setter than a getter?

Can we just have a constructor that sets this to avoid needing to extend the interface?

Reply:
removed the interface changes and added constructor